### ISSUE TITLE
[bug] Conv shape check prints overflowed ints

### ISSUE THREAD

**Issue by SsnL** [pytorch/pytorch#19947]
## 🐛 Bug
```py
In [1]: import torch                                                                                                                                            
   ...: import torch.nn.functional as F                                                                                                                         
   ...:                                                                                                                                                         
   ...:                                                                                                                                                         
   ...: x = torch.randn(1, 10, 1, 28, 28)                                                                                                                       
   ...:                                                                                                                                                         
   ...: w = torch.randn(6, 1, 5, 5)                                                                                                                             
   ...: F.conv2d(x, w)                                                                                                                                          
                                                                                                                                                                
---------------------------------------------------------------------------                                                                                     
RuntimeError                              Traceback (most recent call last)                                                                                     
<ipython-input-1-b22a02dbe070> in <module>                                                                                                                      
      6                                                                                                                                                         
      7 w = torch.randn(6, 1, 5, 5)                                                                                                                             
----> 8 F.conv2d(x, w)                                                                                                                                          
                                                                                                                                                                
RuntimeError: Expected 4-dimensional input for 4-dimensional weight 6 1 5 5 2323362894317625376, but got 5-dimensional input of size [1, 10, 1, 28, 28] instead 
```

**Comment by SsnL** [comment]
caused by 
https://github.com/pytorch/pytorch/blob/8e7750679920e406f77769324d249481db689370/aten/src/ATen/native/Convolution.cpp#L195-L206

**Comment by rotuna** [comment]
Hi, I'd like to try working on this.

**Comment by rotuna** [comment]
@SsnL If I were to understand this code path, what .cpp files would I have to read? Where is globalATenDispatch() defined? I also could not find the definition of the mkldnn_reorder_conv2d_weight function. I could only find it in a generated file. 

How do these files bind Python and C++ code together? Is there any developer documentation other than [this](https://pytorch.org/cppdocs/api/namespace_at.html#namespace-at)?

**Comment by SsnL** [comment]
@rotuna http://blog.ezyang.com/2019/05/pytorch-internals/ is a great guide on the code path! 

For this change, you would need to modify the function I pointed out above, and add a test in `test_nn.py` with `assertRaisesRegex` to check the error message.

Thanks! Let me know if you have any other questions.

**Comment by SsnL** [comment]
fixed in #25827
